pkg/conference/peer_message_processor.go

```diff
@@ -19,32 +19,47 @@ func (c *Conference) processLeftTheCallMessage(participant *Participant, msg pee
 }
 
 func (c *Conference) processNewTrackPublishedMessage(participant *Participant, msg peer.NewTrackPublished) {
-	participant.logger.Infof("Published new track: %s (%s)", msg.TrackID, msg.Layer.String())
+	participant.logger.Infof("Published new track: %s (%v)", msg.TrackID, msg.Layer)
+
+	// If this is a new track, let's add it to the list of published and inform participants.
+	track, found := participant.publishedTracks[msg.TrackID]
+	if !found {
+		layers := []peer.SimulcastLayer{}
+		if msg.Layer != nil {
+			layers = append(layers, *msg.Layer)
+		}
 
-	if track, ok := participant.publishedTracks[msg.TrackID]; !ok {
 		participant.publishedTracks[msg.TrackID] = PublishedTrack{
 			info:   msg.TrackInfo,
-			layers: []peer.SimulcastLayer{msg.Layer},
+			layers: layers,
 		}
-	} else if !funk.Contains(track.layers, msg.Layer) {
-		track.layers = append(track.layers, msg.Layer)
-		participant.publishedTracks[msg.TrackID] = track
+
+		c.resendMetadataToAllExcept(participant.id)
+		return
 	}
 
-	// TODO: Oops, that's not very efficient with the simulcast since it means that when 3
-	// layers are published, we will send 3 times the same metadata.
-	c.resendMetadataToAllExcept(participant.id)
+	// If it's just a new layer, let's add it to the list of layers of the existing published track.
+	if msg.Layer != nil && !funk.Contains(track.layers, *msg.Layer) {
+		track.layers = append(track.layers, *msg.Layer)
+		participant.publishedTracks[msg.TrackID] = track
```

Comment:
Won't all the layers have the same `trackId`?

Reply:
They will, that's why in this case I only modify the `layers` without touching the rest of the fields.